core/src/providers/anthropic.rs

```diff
@@ -75,17 +79,137 @@ impl AnthropicLLM {
             .parse::<Uri>()?)
     }
 
-    async fn generate(
+    async fn streamed_completion(
         &self,
         prompt: &str,
         max_tokens_to_sample: i32,
         temperature: f32,
         top_p: f32,
         top_k: i32,
         stop: &Vec<String>,
+        event_sender: UnboundedSender<Value>,
     ) -> Result<Response> {
-        assert!(self.api_key.is_some());
+        let url = self.uri()?.to_string();
+
+        let mut builder = match es::ClientBuilder::for_url(url.as_str()) {
+            Ok(builder) => builder,
+            Err(e) => {
+                return Err(anyhow!(
+                    "Error creating Anthropic streaming client: {:?}",
+                    e
+                ))
+            }
+        };
+
+        builder = builder.method(String::from("POST"));
+        builder = match builder.header("Content-Type", "application/json") {
+            Ok(builder) => builder,
+            Err(e) => return Err(anyhow!("Error setting header: {:?}", e)),
+        };
+        builder = match builder.header("X-API-Key", self.api_key.clone().unwrap().as_str()) {
+            Ok(builder) => builder,
+            Err(e) => return Err(anyhow!("Error setting header: {:?}", e)),
+        };
+
+        let body = json!({
+            "model": self.id.clone(),
+            "prompt": prompt,
+            "max_tokens_to_sample": max_tokens_to_sample,
+            "temperature": temperature,
+            "stop_sequences": stop.clone(),
+            "top_p": top_p,
+            "top_k": top_k,
+            "stream": true
+        });
+
+        let client = builder
+            .body(body.to_string())
+            .reconnect(
+                es::ReconnectOptions::reconnect(true)
+                    .retry_initial(false)
+                    .delay(Duration::from_secs(1))
+                    .backoff_factor(2)
+                    .delay_max(Duration::from_secs(8))
+                    .build(),
+            )
+            .build();
+
+        let mut stream = client.stream();
+
+        let mut final_response: Option<Response> = None;
+        'stream: loop {
+            match stream.try_next().await {
+                Ok(stream_next) => match stream_next {
+                    Some(es::SSE::Comment(comment)) => {
+                        println!("UNEXPECTED COMMENT {}", comment);
+                    }
+                    Some(es::SSE::Event(event)) => match event.data.as_str() {
+                        "[DONE]" => {
+                            println!("DONE");
+                            break 'stream;
+                        }
+                        _ => {
+                            let response: Response = match serde_json::from_str(event.data.as_str())
```

Comment:
Have you tested openai streaming? Have you tested your code on top of Anthropic?

Anthropic returns the response incrementally so each event has the full text up to the current token being generated. Our API expects to return just the currently generated token. So you'll have to do some processing here to only return the incremental token's text generated.